### ISSUE TITLE
`Column names don't match schema` error when running `Detector1Pipeline`

### ISSUE THREAD

**Issue by ortk95** [spacetelescope/jwst#8832]
Since updating to v1.16.0, we have been getting the error `Column names don't match schema. Schema has {'mt_v2', 'mt_v3'}. Data has set()` when trying to process `*_uncal.fits` files through the `Detector1Pipeline`. We don't have any issues when using v1.15.1 (running identical code on identical files), so this seems to be an issue with the latest release.

```
Traceback (most recent call last):
  File "/Users/ortk2/Dropbox/science/test/data/../pipelines/miri_pipeline.py", line 871, in <module>
    main()
  File "/Users/ortk2/Dropbox/science/test/data/../pipelines/miri_pipeline.py", line 867, in main
    run_pipeline(**vars(parser.parse_args()))
  File "/Users/ortk2/Dropbox/science/test/data/../pipelines/miri_pipeline.py", line 464, in run_pipeline
    pipeline.run(
  File "/Users/ortk2/Dropbox/science/test/pipelines/pipeline.py", line 273, in run
    self.run_step(step)
  File "/Users/ortk2/Dropbox/science/test/pipelines/pipeline.py", line 291, in run_step
    getattr(self, f'run_{step}')(kwargs)
  File "/Users/ortk2/Dropbox/science/test/pipelines/pipeline.py", line 648, in run_stage1
    runmany(
  File "/Users/ortk2/Dropbox/science/test/pipelines/parallel_tools.py", line 66, in runmany
    function(a)
  File "/Users/ortk2/Dropbox/science/test/data/../pipelines/miri_pipeline.py", line 571, in reduction_detector1_fn
    return super().reduction_detector1_fn((path_in, output_dir, kwargs))
  File "/Users/ortk2/Dropbox/science/test/pipelines/pipeline.py", line 657, in reduction_detector1_fn
    Detector1Pipeline.call(
  File "/Users/ortk2/miniconda3/envs/py310-test/lib/python3.10/site-packages/stpipe/step.py", line 680, in call
    config, config_file = cls.build_config(filename, **kwargs)
  File "/Users/ortk2/miniconda3/envs/py310-test/lib/python3.10/site-packages/stpipe/step.py", line 1337, in build_config
    config = cls.get_config_from_reference(input)
  File "/Users/ortk2/miniconda3/envs/py310-test/lib/python3.10/site-packages/stpipe/pipeline.py", line 187, in get_config_from_reference
    with cls._datamodels_open(dataset, asn_n_members=1) as model:
  File "/Users/ortk2/miniconda3/envs/py310-test/lib/python3.10/site-packages/jwst/stpipe/core.py", line 29, in _datamodels_open
    return datamodels.open(init, **kwargs)
  File "/Users/ortk2/miniconda3/envs/py310-test/lib/python3.10/site-packages/stdatamodels/jwst/datamodels/util.py", line 214, in open
    model = new_class(init, **kwargs)
  File "/Users/ortk2/miniconda3/envs/py310-test/lib/python3.10/site-packages/stdatamodels/model_base.py", line 218, in __init__
    asdffile = fits_support.from_fits(init, self._schema, self._ctx,
  File "/Users/ortk2/miniconda3/envs/py310-test/lib/python3.10/site-packages/stdatamodels/fits_support.py", line 720, in from_fits
    known_keywords, known_datas = _load_from_schema(
  File "/Users/ortk2/miniconda3/envs/py310-test/lib/python3.10/site-packages/stdatamodels/fits_support.py", line 639, in _load_from_schema
    mschema.walk_schema(schema, callback)
  File "/Users/ortk2/miniconda3/envs/py310-test/lib/python3.10/site-packages/stdatamodels/schema.py", line 149, in walk_schema
    recurse(schema, [], None, ctx)
  File "/Users/ortk2/miniconda3/envs/py310-test/lib/python3.10/site-packages/stdatamodels/schema.py", line 137, in recurse
    recurse(val, path + [key], combiner, ctx)  # noqa: F821
  File "/Users/ortk2/miniconda3/envs/py310-test/lib/python3.10/site-packages/stdatamodels/schema.py", line 124, in recurse
    if callback(schema, path, combiner, ctx, recurse):  # noqa: F821
  File "/Users/ortk2/miniconda3/envs/py310-test/lib/python3.10/site-packages/stdatamodels/fits_support.py", line 617, in callback
    result = _fits_array_loader(
  File "/Users/ortk2/miniconda3/envs/py310-test/lib/python3.10/site-packages/stdatamodels/fits_support.py", line 563, in _fits_array_loader
    return from_fits_hdu(hdu, schema)
  File "/Users/ortk2/miniconda3/envs/py310-test/lib/python3.10/site-packages/stdatamodels/fits_support.py", line 804, in from_fits_hdu
    data = properties._cast(data, schema)
  File "/Users/ortk2/miniconda3/envs/py310-test/lib/python3.10/site-packages/stdatamodels/properties.py", line 91, in _cast
    val = util.gentle_asarray(val, dtype, allow_extra_columns=allow_extra_columns)
  File "/Users/ortk2/miniconda3/envs/py310-test/lib/python3.10/site-packages/stdatamodels/util.py", line 125, in gentle_asarray
    raise ValueError(
ValueError: Column names don't match schema. Schema has {'mt_v2', 'mt_v3'}. Data has set()
```

If it's relevant, this specific traceback was produced when running MIRI MRS `*_uncal` files from program 1247, using the code in https://github.com/JWSTGiantPlanets/pipelines ([link to the method raising the error](https://github.com/JWSTGiantPlanets/pipelines/blob/2f3c3900149b601da846655c84294ecc268b4b9e/pipeline.py#L657-L659)), with:
- JWST package version: 1.16.1
- Python version: 3.10.15

**Comment by tapastro** [comment]
Hello, thanks for the report! We have a fix coming here: https://github.com/spacetelescope/stdatamodels/pull/329

We're planning to get it merged soon and make a minor release of `stdatamodels` so that installs of `jwst` will pick it up. If you wish to get the fix immediately, you can install stdatamodels from the PR above. Otherwise, I can comment here once the release is out, and you can uninstall/reinstall stdatamodels then.

**Comment by tapastro** [comment]
`stdatamodels=2.1.1` has been released: https://github.com/spacetelescope/stdatamodels/releases/tag/2.1.1

You should be able to `pip uninstall stdatamodels; pip install stdatamodels` to get the update in an existing environment, while new environments should pull the new version by default. Let us know if you encounter further issues!

**Comment by ortk95** [comment]
Thanks - I've updated `stdatamodels` and the `Detector1Pipeline` is now running perfectly!

However, I'm now getting a `CrdsLookupError` for `pastasoss` later on, when running the `Spec3Pipeline`. As before, this error only occurs when I'm using v1.16.0 of the jwst package. It seems like this may be caused by something different, so should I create a new issue for this?

```
Traceback (most recent call last):
  File "/Users/ortk2/Dropbox/science/test/data/../pipelines/miri_pipeline.py", line 871, in <module>
    main()
  File "/Users/ortk2/Dropbox/science/test/data/../pipelines/miri_pipeline.py", line 867, in main
    run_pipeline(**vars(parser.parse_args()))
  File "/Users/ortk2/Dropbox/science/test/data/../pipelines/miri_pipeline.py", line 464, in run_pipeline
    pipeline.run(
  File "/Users/ortk2/Dropbox/science/test/pipelines/pipeline.py", line 273, in run
    self.run_step(step)
  File "/Users/ortk2/Dropbox/science/test/pipelines/pipeline.py", line 291, in run_step
    getattr(self, f'run_{step}')(kwargs)
  File "/Users/ortk2/Dropbox/science/test/data/../pipelines/miri_pipeline.py", line 671, in run_stage3
    super().run_stage3(kwargs)
  File "/Users/ortk2/Dropbox/science/test/pipelines/pipeline.py", line 832, in run_stage3
    runmany(
  File "/Users/ortk2/Dropbox/science/test/pipelines/parallel_tools.py", line 66, in runmany
    function(a)
  File "/Users/ortk2/Dropbox/science/test/pipelines/pipeline.py", line 841, in reduction_spec3_fn
    Spec3Pipeline.call(
  File "/Users/ortk2/miniconda3/envs/py310-test/lib/python3.10/site-packages/stpipe/step.py", line 697, in call
    return instance.run(*args)
  File "/Users/ortk2/miniconda3/envs/py310-test/lib/python3.10/site-packages/jwst/stpipe/core.py", line 100, in run
    result = super().run(*args, **kwargs)
  File "/Users/ortk2/miniconda3/envs/py310-test/lib/python3.10/site-packages/stpipe/step.py", line 522, in run
    self.prefetch(*args)
  File "/Users/ortk2/miniconda3/envs/py310-test/lib/python3.10/site-packages/stpipe/step.py", line 641, in prefetch
    self._precache_references(args[0])
  File "/Users/ortk2/miniconda3/envs/py310-test/lib/python3.10/site-packages/stpipe/pipeline.py", line 266, in _precache_references
    self._precache_references_opened(model)
  File "/Users/ortk2/miniconda3/envs/py310-test/lib/python3.10/site-packages/stpipe/pipeline.py", line 284, in _precache_references_opened
    self._precache_references_opened(contained_model)
  File "/Users/ortk2/miniconda3/envs/py310-test/lib/python3.10/site-packages/stpipe/pipeline.py", line 292, in _precache_references_opened
    self._precache_references_impl(model_or_container)
  File "/Users/ortk2/miniconda3/envs/py310-test/lib/python3.10/site-packages/stpipe/pipeline.py", line 319, in _precache_references_impl
    crds_refs = crds_client.get_multiple_reference_paths(
  File "/Users/ortk2/miniconda3/envs/py310-test/lib/python3.10/site-packages/stpipe/crds_client.py", line 53, in get_multiple_reference_paths
    return _get_refpaths(parameters, tuple(reference_file_types), observatory)
  File "/Users/ortk2/miniconda3/envs/py310-test/lib/python3.10/site-packages/stpipe/crds_client.py", line 65, in _get_refpaths
    bestrefs = crds.getreferences(
  File "/Users/ortk2/miniconda3/envs/py310-test/lib/python3.10/site-packages/crds/core/heavy_client.py", line 127, in getreferences
    best_refs_paths = api.cache_references(
  File "/Users/ortk2/miniconda3/envs/py310-test/lib/python3.10/site-packages/crds/client/api.py", line 848, in cache_references
    wanted = _get_cache_filelist_and_report_errors(bestrefs)
  File "/Users/ortk2/miniconda3/envs/py310-test/lib/python3.10/site-packages/crds/client/api.py", line 891, in _get_cache_filelist_and_report_errors
    raise last_error
crds.core.exceptions.CrdsLookupError: Error determining best reference for 'pastasoss'  =   Unknown reference type 'pastasoss'
```

**Comment by tapastro** [comment]
Sorry, you're getting to experience all of the bugs. 🙃 

Our latest release of `jwst` includes usage of a new reference file type `pastasoss`, which is one of a few changes to CRDS files that are meant only for `jwst>=1.16.0`. Those files are available to users on 1.16, but the CRDS contexts that include them are not selected by default, so as not to change calibrations incorrectly for all users on `jwst<1.16.0`.

Because of this, you'll need to set your context manually. The correct context for 1.16 is `jwst_1295.pmap`, you can set this with the environment variable `CRDS_CONTEXT`.

We have an update coming to CRDS which will select the best context given the `jwst` software version you are running, but it won't be operational for another week or two.

**Comment by ortk95** [comment]
Thanks! It works perfectly with the 1295 context :)
